pkg/kubelet/network/dns/dns.go

```diff
@@ -241,19 +247,99 @@ func (c *Configurer) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
 			return nil, err
 		}
 	}
-	useClusterFirstPolicy := ((pod.Spec.DNSPolicy == v1.DNSClusterFirst && !kubecontainer.IsHostNetworkPod(pod)) || pod.Spec.DNSPolicy == v1.DNSClusterFirstWithHostNet)
-	if useClusterFirstPolicy && len(c.clusterDNS) == 0 {
+	return &runtimeapi.DNSConfig{
+		Servers:  hostDNS,
+		Searches: hostSearch,
+		Options:  hostOptions,
+	}, nil
+}
+
+func (c *Configurer) getPodDNSType(pod *v1.Pod) podDNSType {
+	var useClusterFirstPolicy bool
```

Comment:
This logic seems kinds of hairy but should be simple. Can we reorganize as follows:

```
switch {
  case dnsPolicy == v1.DNSDefault: return podDNSHost
  case dnsPolicy == v1.DNSNone:
    if (/*... alphaenabled*/) { return podDNSNone }
    return podDNSCluster
etc...
```

Reply:
Sure, changed to switch style.